src/irmin-pack/pack.ml

```diff
@@ -151,18 +158,33 @@ struct
     if IO.version block <> current_version then
       Fmt.failwith "invalid version: got %S, expecting %S" (IO.version block)
         current_version;
-    { block; index; lock; dict }
+    { block; index; lock; dict; counter = 1 }
 
   let (`Staged v) =
-    with_cache ~clear ~v:(fun index -> unsafe_v ~index) "store.pack"
+    with_cache ~clear ~valid ~incr_counter
```

Comment:
same here, would be nice to merge `valid` and `incr_counter` into only one function.